### PR TITLE
Another round of DoAfter fixes

### LINKED ISSUE #14289: wall doesnt check if cancelled on doafter
## Description
lets you rapidly make wall

**Reproduction**
1. make girder
2. start adding steel
3. move
4. instantly made

add args.Cancelled check to the doafter handler

### PR DESCRIPTION
**Pull request by keronshb**
## About the PR

closes #14289 

Walls and construction now properly cancel out of a doafter.
Cancelling screwing doors or anything with wires won't prevent further doafters from running anymore.
EscapeInventory can no longer spam and cancels on drop again.
Mopping won't break anymore if cancelled
Meatspikes won't break anymore on use or if cancelled.

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- fix: Walls and construction now properly cancel out of a doafter.
- fix: Cancelling screwing doors or anything with wires won't prevent further doafters from running anymore.
- fix: EscapeInventory can no longer spam and cancels on drop again.
- fix: Mopping won't break anymore if cancelled
- fix: Meatspikes won't break anymore on use or if cancelled.

### CODE REVIEW

File: Content.Server/Resist/CanEscapeInventoryComponent.cs [hunk 1]
```diff
@@ -8,4 +10,9 @@ public sealed class CanEscapeInventoryComponent : Component
     /// </summary>
     [DataField("baseResistTime")]
     public float BaseResistTime = 5f;
+
+    [DataField("isEscaping")]
+    public bool IsEscaping;
```
Comment (ElectroJr) [inline Content.Server/Resist/CanEscapeInventoryComponent.cs:15]: unless do-afters support saving & loading from maps, this shouldn't be a data field

File: Content.Server/Resist/EscapeInventorySystem.cs [hunk 2]
```diff
@@ -82,17 +88,26 @@ private void AttemptEscape(EntityUid user, EntityUid container, CanEscapeInvento
...
 
+        component.CancelToken = null;
+        component.IsEscaping = false;
         args.Handled = true;
     }
 
     private void OnDropped(EntityUid uid, CanEscapeInventoryComponent component, DroppedEvent args)
     {
-        //TODO: Enter cancel logic here
+        component.CancelToken = null;
```
Comment (ElectroJr) [inline Content.Server/Resist/EscapeInventorySystem.cs:110]:
```suggestion
		component.CancelToken?.Cancel();
        component.CancelToken = null;
```

Comment (keronshb) [reply Content.Server/Resist/EscapeInventorySystem.cs:110]: Oop all done and the fix that prevented them from being cancelled

**Comment by AJCM-git** [comment]
test failure

**Comment by TokenStyle** [comment]
This is second PR for DoAfter refactor https://github.com/space-wizards/space-station-14/pull/13225 fixes

**Pull request merged**